src/vs/base/parts/tree/browser/treeView.ts

```diff
@@ -511,10 +512,19 @@ export class TreeView extends HeightMap {
 			vertical: (typeof context.options.verticalScrollMode !== 'undefined' ? context.options.verticalScrollMode : ScrollbarVisibility.Auto),
 			useShadows: context.options.useShadows
 		});
-		this.scrollableElement.onScroll((e) => {
-			this.render(e.scrollTop, e.height, e.scrollLeft, e.width, e.scrollWidth);
-			this._onDidScroll.fire();
-		});
+
+		// {{SQL CARBON EDIT}}
+		if (this.context.dataSource instanceof JobStepsViewDataSource) {
```

Comment:
I don't think it makes sense for the VS Code TreeControl to take a dependency on the SQL Agent JobStepsViewDataSource class.